lib/node_modules/@stdlib/math/base/tools/normhermitepoly/lib/factory.js

```diff
@@ -0,0 +1,58 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2018 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var normhermitepoly = require( './main.js' );
+
+
+// MAIN //
+
+/**
+* Returns a function for evaluating a normalized Hermite polynomial.
+*
+* @param {NonNegativeInteger} n - polynomial degree
+* @returns {Function} function for evaluating a normalized Hermite polynomial
+*
+* @example
+* var polyval = factory( 2 );
+*
+* var v = polyval( 0.5 );
+* // returns -0.75
+*/
+function factory( n ) {
+	return polyval;
+
+	/**
+	* Evaluates a normalized Hermite polynomial.
+	*
+	* @private
+	* @param {number} x - value at which to evaluate a normalized Hermite polynomial
+	* @returns {number} result
+	*/
+	function polyval( x ) {
+		return normhermitepoly( n, x );
```

Comment:
In the `dists` packages, we often have not just used a closure, but a separate inlined implementation that foregoes any of the unnecessary checks. Here, we could return a constant function returning one when `n = 0`, `x` when `n = 1`, and `NaN` when `n` is equal to `NaN`, not an integer, or negative. In this case, the body of the regular `polyval` function could just become

```
if ( isnan( x ) ) { 
  return NaN
}
y2 = 1.0;
y3 = 0.0;
for ( i = n; i > 1; i-- ) {
  y1 = (x*y2) - (i*y3);
  y3 = y2;
  y2 = y1;
}
return (x*y2) - y3;
```

But maybe this is not worth it?

Reply:
The body of the main export would have to stay the same. Only what is returned by the factory function would change.

Reply:
Ah, but okay. I see what you are saying. You are seeing that the checks are not necessary because of how the algorithm evaluates.

Reply:
Actually, no, I believe this is incorrect. When `n=0`, the algo above does not return `1.0`.

Reply:
And when `n` is `NaN`, then your suggestion to simplify the main export does not work.

Reply:
Hmm, maybe I am confused or not expressing myself clearly. In the main export, we have the following:

```js
	if ( isnan( n ) || isnan( x ) || n < 0 || !isint( n ) ) {
		return NaN;
	}
	if ( n === 0 ) {
		// `x` is completely canceled from the expression:
		return 1.0;
	}
	if ( n === 1 ) {
		return x;
	}
```

The returned function from the `factory` method does not have to perform these checks at each iteration (except for `isnan( x )`).

Reply:
Ah, okay. Sorry. I see what you are suggesting now. Yeah, I can make that change.

Reply:
Expressed differently, when the factory method is invoked with `n = 1`, it could return the following function:

```js
function polyval( x ) {
  if ( isnan( x  ) ) { 
    return NaN; 
  }
  return x;
}

etc.

Reply:
And your suggestion makes the `factory` function a bit more "worth it", in comparison to the currying it does now.

Reply:
Well, maybe. Or the compiler does some magic and it's worse performance-wise. :sweat_smile:

Reply:
Done.

Reply:
Interestingly, I am seeing worse performance across various Node versions for the factory method. I am not sure why. Could be that the enclosed function is not as "inline-able", but I don't have much justification for this hypothesis. Not sure if this result persists across engines. In any event, made the changes.